### PR TITLE
Create merge-a-branch-as-a-single-commit.md

### PR DESCRIPTION
**Pull request by elhoucine**

### CODE REVIEW

File: git/merge-a-branch-as-a-single-commit.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+Let's say you want to merge "my-feature" (which contain 10 commits :D ) into develop:
+
+    git checkout develop
+    git merge --squash my-feature
+    git commit -m "Merging my-feature branch"
+
+Now instead of adding all "my-feature" commits to develop we added them all as a single commit.
+
+###### I don't recommand using this always but sometimes it's useful to keep history clean.
```
Comment (bronislav) [inline git/merge-a-branch-as-a-single-commit.md]: recomm**e**nd

File: git/merge-a-branch-as-a-single-commit.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Merge all changes from another branch as a single commit:
+
+Let's say you want to merge "my-feature" (which contain 10 commits :D ) into develop:
+
+    git checkout develop
+    git merge --squash my-feature
```
Comment (zamith) [inline git/merge-a-branch-as-a-single-commit.md]: I think `rebase -i` is even better, as it is more flexible in what you can do with the commits.

File: git/merge-a-branch-as-a-single-commit.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    git commit -m "Merging my-feature branch"
+
+Now instead of adding all "my-feature" commits to develop we added them all as a single commit.
+Now instead of adding all "my-feature" commits to develop we added them as a single one.
```
Comment (zamith) [inline git/merge-a-branch-as-a-single-commit.md]: I believe this line is duplicated.

File: git/merge-a-branch-as-a-single-commit.md (hunk 2 above)
Comment (elhoucine) [inline git/merge-a-branch-as-a-single-commit.md]: Hello zamith, As someone who getting started with git I found git merge --squash simpler for this use case. 

If you'd like you can add an exemple of rebase -i as another way to do it.

Thanks

File: git/merge-a-branch-as-a-single-commit.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+# Merge all changes from another branch as a single commit:
+
+Let's say you want to merge "my-feature" (which contain 10 commits :D ) into develop:
```
Comment (croaky) [inline git/merge-a-branch-as-a-single-commit.md]: Maybe put backticks around `my-feature` and `develop`?

**Comment by croaky** [comment]
The `--squash` flag is cool! Hadn't seen that before. TIL!

One small downside to this approach might be that GitHub won't automatically mark the pull request as merged. Out [Git merge protocol](https://github.com/thoughtbot/guides/tree/master/protocol/git#merge) has generally been rebase on the branch, force push it up to origin, then merge. This `--squash` flag would be very handy in other, non-PR situations, though. Good to know.

Somewhat related: `git up`, `git merge-branch` are a couple nice aliases to save a step or two. We also [configure `merge` to always `ff-only`](https://github.com/thoughtbot/dotfiles/blob/master/gitconfig#L25).

File: git/merge-a-branch-as-a-single-commit.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Merge all changes from another branch as a single commit:
+
+Let's say you want to merge `my-feature` (which contain 10 commits :D ) into `develop`:
```
Comment (mike-burns) [inline git/merge-a-branch-as-a-single-commit.md]: Drop the `:D `, please.

File: git/merge-a-branch-as-a-single-commit.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+    git checkout develop
+    git merge --squash my-feature // Will stage new files for commit.
```
Comment (mike-burns) [inline git/merge-a-branch-as-a-single-commit.md]: Since this is sh, use `#` instead of `//`.

File: git/merge-a-branch-as-a-single-commit.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+    git commit -m "Merging my-feature branch"
```
Comment (mike-burns) [inline git/merge-a-branch-as-a-single-commit.md]: Using `-m` is an anti-pattern, but I can see its convenience here in this demo. Is there a better way to show this?

Comment (arturoherrero) [inline git/merge-a-branch-as-a-single-commit.md]: An anti-pattern? If you don't need to write a description, I think that is something perfectly valid.

File: git/merge-a-branch-as-a-single-commit.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+# Merge all changes from another branch as a single commit:
+
+Let's say you want to merge `my-feature` (which contain 10 commits :D ) into `develop`:
+
+    git checkout develop
```
Comment (mike-burns) [inline git/merge-a-branch-as-a-single-commit.md]: Is `develop` a reference to git-flow?

Comment (elhoucine) [inline git/merge-a-branch-as-a-single-commit.md]: Yes.

**Comment by mike-burns** [comment]
Does this create merge commits on master? If so, this is back to: something that someone has legitimately learned vs. a thing that we don't want to teach.

**Pull request closed without merging**